pkg/executor/push.go

```diff
@@ -126,7 +126,7 @@ func pushLayerToCache(opts *config.KanikoOptions, cacheKey string, layer v1.Laye
 	if err != nil {
 		return errors.Wrap(err, "appending layer onto empty image")
 	}
-	return DoPush(empty, &config.KanikoOptions{
-		Destinations: []string{cache},
-	})
+	cacheOpts := opts
+	cacheOpts.Destinations = []string{cache}
```

Comment:
This mutates opts, could we do it by creating a new object?

Reply:
sorry, I think I've fixed it

Reply:
I think this should work from this [test](https://play.golang.org/p/Nld2igwDRAx)